pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,198 @@
+import pymc as pm
+import numpy as np
+import pytensor.tensor as at
+import pytensor
+
+from pymc.distributions.distribution import (
+    Distribution,
+    Discrete,
+    SymbolicRandomVariable,
+    _moment,
+)
+
+from pymc.logprob.abstract import _logprob
+from pymc.pytensorf import intX
+from pytensor.tensor import TensorVariable
+from pytensor.tensor.random.op import RandomVariable
+from pytensor.graph.basic import Node
+
+from pymc.distributions.shape_utils import (
+    _change_dist_size,
+    get_support_shape_1d,
+)
+
+
+def validate_transition_matrix(P):
+    """
+    Checks that P is a valid transition matrix
+    """
+
+    # TODO: Can this eval be avoided?
+    n, k = P.shape.eval()
+    if n != k:
+        raise ValueError(f'P must be square, found shape ({n}, {k})')
+
+    row_sums_all_one = at.allclose(P.sum(axis=1), 1.0)
+    if not row_sums_all_one.eval():
+        raise ValueError('All rows of P must sum to 1.')
+
+
+class DiscreteMarkovChainRV(SymbolicRandomVariable):
+    default_output = 1
+    _print_name = ('DiscreteMC', '\\operatorname{DiscreteMC}')
+
+    def update(self, node: Node):
+        # TODO: Do I need this?
```

Comment:
Yes, otherwise the categorical in the scan won't be properly seeded

Reply:
Thanks!